### ISSUE TITLE
Cannot Get Test Project to Load Changes in a SG-Based Project

### ISSUE THREAD

**Issue by JasonBock** [dotnet/roslyn#69423]
**Version Used**: 
VS 17.6.5
.NET 7

**Steps to Reproduce**:

1. Clone the repo here: https://github.com/JasonBock/Rocks
2. Build the solution

**Expected Behavior**:

I'd expect it to build.

**Actual Behavior**:

I'm getting errors like this:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS8763	A method marked [DoesNotReturn] should not return.	Rocks.IntegrationTests	C:\Users\jason\source\repos\Rocks\src\Rocks.IntegrationTests\Rocks\Rocks.RockCreateGenerator\Rocks.IntegrationTests.UsesDoesNotReturn_Rock_Create.g.cs	119	Active
```

To give more insight into this....I just made a change in my SG that was based on finding members marked with `[DoesNotReturn]`. My tests defined in `DoesNotReturnGeneratorTests` in `Rocks.Tests` run just fine, and demonstrates that my change is affecting the generated output (specifically, my override of `IntMethod()` no longer returns, and will always throw an exception).

However, my `Rocks.IntegrationTests` project keeps giving those `CS8763` errors. It's because I added tests in the `DoesNotReturnTests` class. It **should** pick up that the gen'd code is correct, but it's not.

I can run `dotnet test` from Terminal in the directory that contains `Rocks.IntegrationTests`, and everything builds, all tests pass.

I've been maintaining this solution for years. I know VS won't reload changes from the `Rocks` project, so I'll just do a "clean", close VS, restart VS, and all is well. Now, for some reason, VS just will not pick up my changes. I've even rebooted Windows to no avail.

I would be kind of surprised if this can be reproduced on another machine. My guess is that something locally (maybe with a recent VS update?) is causing the problem.

**Comment by CyrusNajmabadi** [comment]
What behavior do you see on the latest VS previews?

**Comment by JasonBock** [comment]
17.7.0 Preview 6 has the same behavior.

**Comment by JasonBock** [comment]
FWIW I just updated to 17.7 and 17.8 Preview 1, both still fail. Running `dotnet test` from the `src` folder works just fine.

I'm going to make **absolutely sure** there isn't something odd with the project references when I run this in VS, but given that `dotnet test` works....I don't know what's going on.

**Comment by JasonBock** [comment]
OK, I had a hunch, and I just confirmed it. Totally my fault, not VS or Roslyn :)

I have an extension method for `IMethodSymbol` called `IsMarkedWithDoesNotReturn()`. The way I currently do it, I try to load a `Type` based on the name of `DoesNotReturnAttribute`. In retrospect, this was totally the **wrong** way to do it. But in my Rocks.Tests project, this worked. In the Rocks.IntegrationTests project, the type isn't found, so the code isn't generated correctly. I changed this to just look at the name of the attribute for comparison purposes, and now everything works. I'm guessing when `dotnet test` is invoked, the type is discovered, which is why it was working from the CLI.

I'll close this issue.
